File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/ResistorColor.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp;

public static class ResistorColor
{
    public enum BandColor
    {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white
    }
    
    public static int ColorCode(string color)
    {
        Enum.TryParse(color, true, out BandColor result);
        return (int)result;
    }

    public static string[] Colors()
    {
        return Enum.GetNames(typeof(BandColor));
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/ResistorColor.java
package exercism;

class ResistorColor {
    private static final String[] COLORS = {
            "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
    };

    int colorCode(String color) {
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        throw new IllegalArgumentException("Unknown color: " + color);
    }

    String[] colors() {
        return COLORS;
    }
}
